src/crawlee/_autoscaling/system_status.py

```diff
@@ -182,17 +181,19 @@ def _is_sample_overloaded(self, sample: list[Snapshot], threshold: float) -> Loa
         if len(sample) == 1:
             return LoadRatioInfo(limit_ratio=threshold, actual_ratio=float(sample[0].is_overloaded))
 
-        weights, values = [], []
+        overloaded_time = 0.0
+        non_overloaded_time = 0.0
```

Comment:
I did halfway change. I think it is equally readable If I track both overloaded_time and total_time then I have to do some extra sum on each cycle. Super insignificant, but since I already think both approaches are kind of the same, this tips the scales for my decision.